hazelcast/src/main/java/com/hazelcast/spi/impl/sequence/CallIdSequenceWithBackpressure.java

```diff
@@ -94,13 +94,20 @@ private boolean hasSpace() {
         return longs.get(INDEX_HEAD) - longs.get(INDEX_TAIL) < maxConcurrentInvocations;
     }
 
-    private void waitForSpace() throws TimeoutException {
-        long deadline = System.nanoTime() + backoffTimeoutNanos;
+    private void waitForSpace() {
+        if (backoffTimeoutNanos <= 0) {
```

Comment:
How can backoffTimeoutNanos be smaller than 0? It is configured with a negative value, it is a configuration error and should be taken care of by the constructor.

Reply:
I am adding a new class called CallIdSequenceWithBackpressureFailFast, which does not have any wait logic. Also added positive number check for the parameters of the  CallIdSequenceWithBackpressure constructor.